lms/static/scripts/frontend_apps/components/test/ContentSelector-test.js

```diff
@@ -352,6 +369,137 @@ describe('ContentSelector', () => {
     });
   });
 
+  describe('OneDrive picker', () => {
+    function clickOneDriveButton(wrapper) {
+      const btn = wrapper.find('LabeledButton[data-testid="onedrive-button"]');
+      interact(wrapper, () => {
+        btn.props().onClick();
+      });
+    }
+
+    beforeEach(() => {
+      const picker = FakeGooglePickerClient();
+      picker.showPicker.resolves({
+        id: 'doc1',
+        url: 'https://files.google.com/doc1',
+      });
+      picker.enablePublicViewing.resolves();
+
+      // Silence errors logged if showing OneDrive Picker fails.
+      sinon.stub(console, 'error');
+    });
+
+    afterEach(() => {
+      console.error.restore();
+    });
+
+    it('skips initialization of OneDrive Picker client if parameters not provided', () => {
+      fakeConfig.filePicker.microsoftOneDrive = {
+        enabled: true,
+      };
+      renderContentSelector();
+
+      assert.notCalled(FakeOneDrivePickerClient);
+    });
+
+    it("doesn't show the OneDrive button if option is disabled", () => {
+      fakeConfig.filePicker.microsoftOneDrive = {
+        enabled: false,
```

Comment:
You might want to include the parameters here to prove that the option is not shown because of the `enabled` value, rather than because the other parameters are missing. You could also consider combining this with the previous test as part of a parametrized test.

Reply:
I have set `enable` false while leaving the `clientId` and `redirectURI` defined.